src/Appwrite/Migration/Migration.php

```diff
@@ -0,0 +1,88 @@
+<?php
+
+namespace Appwrite\Migration;
+
+use Appwrite\Database\Document;
+use Appwrite\Database\Database;
+use Utopia\CLI\Console;
+use Utopia\Exception;
+
+abstract class Migration
+{
+    protected \PDO $db;
+
+    protected int $limit = 30;
+
+    protected Document $project;
+    protected Database $projectDB;
+
+    /**
+     * Migration constructor.
+     */
+    public function __construct(\PDO $db)
+    {
+        $this->db = $db;
+    }
+
+    /**
+     * Set project for migration.
+     */
+    public function setProject(Document $project, Database $projectDB): Migration
+    {
+        $this->project = $project;
+        $this->projectDB = $projectDB;
+        $this->projectDB->setNamespace('app_' . $project->getId());
+        return $this;
+    }
+
+    /**
+     * Iterates through every document.
+     * 
+     * @param callable(Document):Document $callback
+     */
```

Comment:
Probably should just be `@param callable $callback`

Reply:
But this doesnt imply the needed argument and needed return, right?

Reply:
I found this syntax [here](https://psalm.dev/docs/annotating_code/type_syntax/callable_types/). 🙂

Reply:
Honestly, I have no idea, never seen that syntax before. 😕